### PR TITLE
Add information about build variants to contributors guide 

### PR DESCRIPTION
**Pull request by khushishikhu**
closes odk-x/tool-suite-X#229

addresses odk-x/tool-suite-X#229

#### What is included in this PR?
This PR includes the documentation for the contributor's guide about build variants which has been asked in the issue.

#### What problems did you encounter?
I am not sure of one that is this the correct location where this documentation to be written. Please correct me if I was wrong in identifying it.

### CODE REVIEW

File: odkx-src/build-scripts.rst [hunk 1]
```diff
@@ -148,7 +148,27 @@ To run the UI tests:
 
       ./gradlew clean testSnapshotBasicDebug connectedSnapshotBasicDebugAndroidTest
 
+Build Variants
+~~~~~~~~~~~~~~~~~~
+
```
Comment (wbrunette) [inline odkx-src/build-scripts.rst:153]: It would be good to start off with an introduction of why we use build variants.
"ODK-X has a modular framework design with inter-dependencies between various ODK-X tools. To manage accepting code contributions, bug fixes, and other enhancements that might effect other tools in a unforeseen way ODK-X uses a 3 staged release workflow that involves 3 branches.

- development: the branch where new development, upgrades, and features are contributed and tested
- demo: This branch is the last stable version of development. This provides a staging area used for testing before moving to an official release. This is where preview releases are staged.
- master: The current stable release.
"

Since the branches are a different stages of development the dependency libraries are at different stages of development. To make sure you are building against the correct dependencies you need to either checkout a local copy of the dependencies using the same branch name in the same directory OR you can adjust the build variant to match the dependencies causing Gradle and Android Studio to fetch the correct dependencies when compiling.

File: odkx-src/build-scripts.rst [hunk 2]
```diff
@@ -148,7 +148,27 @@ To run the UI tests:
...
+you can configure build variants to create different versions of your app from a single project, and how to properly manage your dependencies and signing configurations. Each build variant represents a different version of your app that You can configure a dependency for a specific build variant. For more information refer to https://developer.android.com/studio/build/build-variants
```
Comment (wbrunette) [inline odkx-src/build-scripts.rst:154]: I am confused by what you mean by different versions. While it's a true statement a version number does not correspond to a build variant (it could, but it does not). You can also probably remove signing configurations as people do not need to worry about that.

File: odkx-src/build-scripts.rst [hunk 3]
```diff
@@ -148,7 +148,27 @@ To run the UI tests:
...
+
+The Build variant corresponding to the source branches:
```
Comment (wbrunette) [inline odkx-src/build-scripts.rst:156]: Add in a note about the proper dependency libraries for the branch.

File: odkx-src/build-scripts.rst [hunk 4]
```diff
@@ -148,7 +148,27 @@ To run the UI tests:
...
+
+1. **development**:  **snapshotBasicRelease** is one of the build variant corresponds to development branch in Android studio. All the required changes and forks are done from this branch in which the new development, upgrades and features should be contributed and tested.
```
Comment (wbrunette) [inline odkx-src/build-scripts.rst:158]: It's not limit to Android Studio, it includes the command line tool Gradle. I would drop the word Android studio here and just have a section below telling people how to change the build variant in a sub-section about Android Studio after this list.

File: odkx-src/build-scripts.rst [hunk 5]
```diff
@@ -148,7 +148,27 @@ To run the UI tests:
...
+2. **demo**:  **demoBasicRelease** here is the build variant that corresponds do demo branch in Android studio. This is basically the preview release of an application before launching the official release. Which can bu used by project maintainers for testing out the application if it is not creating any errors.
```
Comment (wbrunette) [inline odkx-src/build-scripts.rst:159]: Again I would drop Android Studio from here as it's kind of a separate thing as this works outside of Android Studio.

File: odkx-src/build-scripts.rst [hunk 6]
```diff
@@ -148,7 +148,27 @@ To run the UI tests:
...
+3. **master**:  **masterBasicRlease** is the official and stable release of an application, that is to be used globally.
```
Comment (wbrunette) [inline odkx-src/build-scripts.rst:160]: All 3 of the variants are used globally, so saying used globally confuses me. Are you trying to say that this build variant will pull the dependency libraries for the latest release of ODK-X.

**Review by wbrunette — CHANGES_REQUESTED** [review]
@khushishikhu Great start!

Please make the requested changes and then hit the refresh/recycle button in review and the system will notify us that the pull request is ready for review again.

File: odkx-src/build-scripts.rst (hunk 4 above)
Comment (khushishikhu) [reply odkx-src/build-scripts.rst:158]: okay sure I got it. But I have written the steps to change the build variant after this list.

**Comment by khushishikhu** [comment]
@wbrunette Thank you!! for reviewing my PR. I have made all the changes in all those places  you asked for to do. Kindly check the changes now and let me know what more it requires.

File: odkx-src/build-scripts.rst [hunk 7]
```diff
@@ -148,7 +148,41 @@ To run the UI tests:
...
+
+you can configure build variants to create different versions of your app from a single project, and how to properly manage your dependencies and signing configurations. Each build variant represents a different version of your app that You can configure a dependency for a specific build variant. For more information refer to https://developer.android.com/studio/build/build-variants
+
+
+-development: the branch where new development, upgrades, and features are contributed and tested
+-demo: This branch is the last stable version of development. This provides a staging area used for testing before moving to an official release. This is where preview releases are staged.
+-master: The current stable release.
+
+Since the branches are a different stages of development the dependency libraries are at different stages of development. To make sure you are building against the correct dependencies you need to either checkout a local copy of the dependencies using the same branch name in the same directory OR you can adjust the build variant to match the dependencies causing Gradle and Android Studio to fetch the correct dependencies when compiling.
```
Comment (linl33) [inline odkx-src/build-scripts.rst:164]: ~~are a different~~
are at different

File: odkx-src/build-scripts.rst [hunk 8]
```diff
@@ -148,7 +148,41 @@ To run the UI tests:
 
       ./gradlew clean testSnapshotBasicDebug connectedSnapshotBasicDebugAndroidTest
 
+Build Variants
+~~~~~~~~~~~~~~~~~~
+
+
+ODK-X has a modular framework design with inter-dependencies between various ODK-X tools. To manage accepting code contributions, bug fixes, and other enhancements that might effect other tools in a unforeseen way ODK-X uses a 3 staged release workflow that involves 3 branches.
```
Comment (linl33) [inline odkx-src/build-scripts.rst:155]: ~~in a unforeseen~~
in an unforeseen

Comment (linl33) [inline odkx-src/build-scripts.rst:155]: ~~might effect~~
might affect

File: odkx-src/build-scripts.rst [hunk 9]
```diff
@@ -148,7 +148,41 @@ To run the UI tests:
...
+
+you can configure build variants to create different versions of your app from a single project, and how to properly manage your dependencies and signing configurations. Each build variant represents a different version of your app that You can configure a dependency for a specific build variant. For more information refer to https://developer.android.com/studio/build/build-variants
+
+
+-development: the branch where new development, upgrades, and features are contributed and tested
```
Comment (linl33) [inline odkx-src/build-scripts.rst:160]: Please check the formatting for this paragraph. Everything is displayed on a single line.

File: odkx-src/build-scripts.rst [hunk 10]
```diff
@@ -148,7 +148,41 @@ To run the UI tests:
...
+
+  On changing any build variant that will run in the gradle command line as it runs for other vanriants, you are just aware of which version is presenting of that application. That has different functionalities in every version as all those are made for different purpose.
+
+Steps to change build variants in Android studio:
+
+1. To change the build variant Android Studio uses, select **Build** > **Select Build Variant** in the menu bar.
+2. The Build Variants panel has two columns: **Module** and **Active Build Variant**. The **Active Build Variant** value for the module determines which build variant the IDE deploys to your connected device and is visible in the editor.
 
+.. image:: /img/survey-using/build-variants.*
+    :alt: Survey Form List
```
Comment (linl33) [inline odkx-src/build-scripts.rst:184]: Please update the `alt` to reflect this image.

File: odkx-src/build-scripts.rst [hunk 11]
```diff
@@ -148,7 +148,41 @@ To run the UI tests:
...
+
+.. note::
+
+  The Build variant corresponding to the source branches:
+
+  1. **development**:  **snapshotBasicRelease** is one of the build variant corresponds to development branch. All the required changes and forks are done from this branch in which the new development, upgrades and features should be contributed and tested.
+  2. **demo**:  **demoBasicRelease** here is the build variant that corresponds do demo branch . This is basically the preview release of an application before launching the official release. Which can bu used by project maintainers for testing out the application if it is not creating any errors.
+  3. **master**:  **masterBasicRlease** is the official and stable release of an application, this is the latest release of odkx application.
+
+  On changing any build variant that will run in the gradle command line as it runs for other vanriants, you are just aware of which version is presenting of that application. That has different functionalities in every version as all those are made for different purpose.
```
Comment (linl33) [inline odkx-src/build-scripts.rst:176]: Could you rephrase these 2 sentences? It's not clear what you're trying to say here.

File: odkx-src/build-scripts.rst [hunk 12]
```diff
@@ -148,7 +148,41 @@ To run the UI tests:
...
+
+Each build variant represents a different version of your app that You can configure a dependency for a specific build variant. https://developer.android.com/studio/build/build-variants
+
+.. note::
+
+  The Build variant corresponding to the source branches:
+
+  1. **development**:  **snapshotBasicRelease** is one of the build variant corresponds to development branch. All the required changes and forks are done from this branch in which the new development, upgrades and features should be contributed and tested.
+  2. **demo**:  **demoBasicRelease** here is the build variant that corresponds do demo branch . This is basically the preview release of an application before launching the official release. Which can bu used by project maintainers for testing out the application if it is not creating any errors.
+  3. **master**:  **masterBasicRlease** is the official and stable release of an application, this is the latest release of odkx application.
```
Comment (linl33) [inline odkx-src/build-scripts.rst:174]: ~~odkx~~
ODK-X

File: odkx-src/build-scripts.rst [hunk 13]
```diff
@@ -148,7 +148,41 @@ To run the UI tests:
...
+
+Each build variant represents a different version of your app that You can configure a dependency for a specific build variant. https://developer.android.com/studio/build/build-variants
+
+.. note::
+
+  The Build variant corresponding to the source branches:
+
+  1. **development**:  **snapshotBasicRelease** is one of the build variant corresponds to development branch. All the required changes and forks are done from this branch in which the new development, upgrades and features should be contributed and tested.
+  2. **demo**:  **demoBasicRelease** here is the build variant that corresponds do demo branch . This is basically the preview release of an application before launching the official release. Which can bu used by project maintainers for testing out the application if it is not creating any errors.
```
Comment (linl33) [inline odkx-src/build-scripts.rst:173]: ~~corresponds do~~
corresponds to

~~can bu~~
can be

**Review by linl33 — CHANGES_REQUESTED** [review]
Thanks for getting this documented.

**Comment by linl33** [comment]
@khushishikhu It looks like your Git LFS isn't configured correctly. The image you added isn't tracked in Git LFS. Could you try to get the image in Git LFS and then **force push** to remove the copy of the image tracked in regular Git.

**Comment by khushishikhu** [comment]
@linl33 I have made all the required changes you asked for and also tracked the file in git LFS. Please review it and let me know what else needs to be corrected more.

**Comment by wbrunette** [comment]
@khushishikhu please make sure your pull requests passes the build checks as we cannot merge something that won't build properly.

**Comment by wbrunette** [comment]
@khushishikhu it looks like spelling errors so it shouldn't be too hard to fix :)

**Comment by khushishikhu** [comment]
@wbrunette I have fixed the errors. The checks are passing now. Please review it now

**Comment by wbrunette** [comment]
@khushishikhu I just took a look and some of the grammar issues that @linl33 pointed out still exist. Please make all the corrections listed.

**Comment by khushishikhu** [comment]
sorry!! @wbrunette I don't know why the changes I have done was not committed but I have done again the suggested changes and reviewed also. Please check if it is correct now.

File: odkx-src/build-scripts.rst [hunk 14]
```diff
@@ -148,7 +148,38 @@ To run the UI tests:
...
+
+  On changing any build variant that will run in the gradle command line as it runs for other variants, you are just aware of which version is presenting of that application. That has different functionalities in every version as all those are made for different purpose.
+
+Steps to change build variants in Android studio:
+
+1. To change the build variant Android Studio uses, select **Build** > **Select Build Variant** in the menu bar.
+2. The Build Variants panel has two columns: **Module** and **Active Build Variant**. The **Active Build Variant** value for the module determines which build variant the IDE deploys to your connected device and is visible in the editor.
 
+.. image:: /img/survey-using/build-variants.*
+    :alt: build-variants
```
Comment (linl33) [inline odkx-src/build-scripts.rst:181]: The alt text should be a textual description of the image. See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#attr-alt

File: odkx-src/build-scripts.rst [hunk 15]
```diff
@@ -148,7 +148,38 @@ To run the UI tests:
...
+
+.. note::
+
+  The Build variant corresponding to the source branches:
+
+  1. **development**:  **snapshotBasicRelease** is one of the build variant corresponds to development branch. All the required changes and forks are done from this branch in which the new development, upgrades and features should be contributed and tested.
+  2. **demo**:  **demoBasicRelease** here is the build variant that corresponds to demo branch . This is basically the preview release of an application before launching the official release. Which can be used by project maintainers for testing out the application if it is not creating any errors.
+  3. **master**:  **masterBasicRlease** is the official and stable release of an application, this is the latest release of ODK-X application.
+
+  On changing any build variant that will run in the gradle command line as it runs for other variants, you are just aware of which version is presenting of that application. That has different functionalities in every version as all those are made for different purpose.
```
Comment (linl33) [inline odkx-src/build-scripts.rst:173]: It's still not very clear what you're trying to say here. Are you trying to say that the variant selected in Android Studio doesn't affect the Gradle command line?

File: odkx-src/build-scripts.rst [hunk 16]
```diff
@@ -148,7 +148,38 @@ To run the UI tests:
...
+
+Each build variant represents a different version of your app that You can configure a dependency for a specific build variant. https://developer.android.com/studio/build/build-variants
+
+.. note::
+
+  The Build variant corresponding to the source branches:
+
+  1. **development**:  **snapshotBasicRelease** is one of the build variant corresponds to development branch. All the required changes and forks are done from this branch in which the new development, upgrades and features should be contributed and tested.
+  2. **demo**:  **demoBasicRelease** here is the build variant that corresponds to demo branch . This is basically the preview release of an application before launching the official release. Which can be used by project maintainers for testing out the application if it is not creating any errors.
+  3. **master**:  **masterBasicRlease** is the official and stable release of an application, this is the latest release of ODK-X application.
```
Comment (linl33) [inline odkx-src/build-scripts.rst:171]: ~~masterBasicRlease~
masterBasicRelease

**Review by linl33 — CHANGES_REQUESTED** [review]
Looks like your image still isn't in Git LFS. You can check the files track by Git LFS using `git lfs ls-files`

File: odkx-src/build-scripts.rst (hunk 15 above)
Comment (khushishikhu) [reply odkx-src/build-scripts.rst:173]: @linl33  Are  you asking for the line 173 ?

Comment (linl33) [reply odkx-src/build-scripts.rst:173]: yes 173

File: odkx-src/build-scripts.rst [hunk 17]
```diff
@@ -148,14 +148,45 @@ To run the UI tests:
...
+
+Since the branches are at different stages of development the dependency libraries are at different stages of development. To make sure you are building against the correct dependencies you need to either checkout a local copy of the dependencies using the same branch name in the same directory OR you can adjust the build variant to match the dependencies causing Gradle and Android Studio to fetch the correct dependencies when compiling.
+
+Each build variant represents a different version of your app that You can configure a dependency for a specific build variant. https://developer.android.com/studio/build/build-variants
+
+.. note::
+
+  The Build variant corresponding to the source branches:
+
+  1. **development**:  **snapshotBasicRelease** is one of the build variant corresponds to development branch. All the required changes and forks are done from this branch in which the new development, upgrades, and features should be contributed and tested.
```
Comment (wbrunette) [inline odkx-src/build-scripts.rst:169]: I would drop "All the required changes and forks are done from this branch. Leave it as "this branch is where the new development, upgrades, and features" but change to *are* instead of should be.

Comment (wbrunette) [inline odkx-src/build-scripts.rst:169]: _is one of the build variant corresponds to development branch_

Change to "is the build variant that corresponds to a release build of the development branch. **snapshotBasicDebug** is the build variant that corresponds to a debug build of the development branch.

File: odkx-src/build-scripts.rst [hunk 18]
```diff
@@ -148,14 +148,45 @@ To run the UI tests:
...
+  2. **demo**:  **demoBasicRelease** here is the build variant that corresponds to demo branch . This is the preview release of an application before launching the official release. Which can be used by project maintainers for testing out the application if it is not creating any errors.
```
Comment (wbrunette) [inline odkx-src/build-scripts.rst:170]: *demoBasicRelease** here is the build variant that corresponds to demo branch 

CHANGE TO: "**demoBasicRelease**is the build variant that corresponds to a release build of the demo branch. **demoBasicDebug** is the build variant that corresponds to a debug build of the demo branch. 

(same should be done for all 3)

File: odkx-src/build-scripts.rst [hunk 19]
```diff
@@ -148,14 +148,45 @@ To run the UI tests:
...
+  3. **master**:  **masterBasicRelease** is the official and stable release of an application, this is the latest release of ODK-X application.
```
Comment (wbrunette) [inline odkx-src/build-scripts.rst:171]: *demoBasicRelease** here is the build variant that corresponds to demo branch 

CHANGE TO: "**masterBasicRelease**is the build variant that corresponds to a release build of the master branch. **masterBasicDebug** is the build variant that corresponds to a debug build of the master branch. 

(same should be done for all 3)

**Review by wbrunette — CHANGES_REQUESTED** [review]

**Comment by khushishikhu** [comment]
@wbrunette I have done the suggested changes. Please have a look now

File: odkx-src/build-scripts.rst [hunk 20]
```diff
@@ -148,14 +148,45 @@ To run the UI tests:
...
+
+.. note::
+
+  The Build variant corresponding to the source branches:
+
+  1. **development**:  **snapshotBasicRelease** is the build variant corresponds to a release build of the development branch. **snapshotBasicDebug** is the build variant that corresponds to a debug build of the development branch. This branch is where the new development, upgrades, and features are contributed and tested.
+  2. **demo**:  **demoBasicRelease** is the build variant that corresponds to release build of the demo branch. **demoBasicDebug** is the build variant that corresponds to a debug build of the demo branch. This is the preview release of an application before launching the official release. Which can be used by project maintainers for testing out the application if it is not creating any errors.
+  3. **master**:  **masterBasicRelease** is the build variant that corresponds to a release build of the master branch. **masterBasicDebug** is the build variant that corresponds to a debug build of the master branch.It is the official and stable release of an application, this is the latest release of ODK-X application.
```
Comment (wbrunette) [inline odkx-src/build-scripts.rst:171]: Missing a space

**Review by wbrunette — COMMENTED** [review]
@elmps2018 this is close to be ready, as a non expert can you give it read and see if it makes sense to you. If you ask questions in your review it might help to clarify something that is not clear to a new user.

File: odkx-src/build-scripts.rst [hunk 21]
```diff
@@ -148,14 +148,45 @@ To run the UI tests:
...
+
+- development: the branch where new development, upgrades, and features are contributed and tested
+- demo: This branch is the last stable version of development. This provides a staging area used for testing before moving to an official release. This is where preview releases are staged.
+- master: The current stable release.
+
+Since the branches are at different stages of development the dependency libraries are at different stages of development. To make sure you are building against the correct dependencies you need to either checkout a local copy of the dependencies using the same branch name in the same directory OR you can adjust the build variant to match the dependencies causing Gradle and Android Studio to fetch the correct dependencies when compiling.
+
+Each build variant represents a different version of your app that You can configure a dependency for a specific build variant. https://developer.android.com/studio/build/build-variants
```
Comment (elmps2018) [inline odkx-src/build-scripts.rst:163]: This sentence does not quite make sense. Does it mean "Each build variant represents a different version of your app that you can configure a dependency for"?

File: odkx-src/build-scripts.rst [hunk 22]
```diff
@@ -148,14 +148,45 @@ To run the UI tests:
...
+
+.. note::
+
+  The Build variant corresponding to the source branches:
```
Comment (elmps2018) [inline odkx-src/build-scripts.rst:167]: minor edit -- build should be lower case

File: odkx-src/build-scripts.rst [hunk 23]
```diff
@@ -148,14 +148,45 @@ To run the UI tests:
...
+
+  1. **development**:  **snapshotBasicRelease** is the build variant corresponds to a release build of the development branch. **snapshotBasicDebug** is the build variant that corresponds to a debug build of the development branch. This branch is where the new development, upgrades, and features are contributed and tested.
```
Comment (elmps2018) [inline odkx-src/build-scripts.rst:169]: minor grammar: "is the build variant that corresponds" (that missing)

File: odkx-src/build-scripts.rst [hunk 24]
```diff
@@ -148,14 +148,45 @@ To run the UI tests:
...
+  2. **demo**:  **demoBasicRelease** is the build variant that corresponds to release build of the demo branch. **demoBasicDebug** is the build variant that corresponds to a debug build of the demo branch. This is the preview release of an application before launching the official release. Which can be used by project maintainers for testing out the application if it is not creating any errors.
```
Comment (elmps2018) [inline odkx-src/build-scripts.rst:170]: to a release build (missing a)

Comment (elmps2018) [reply odkx-src/build-scripts.rst:170]: "Which can..." should be "This branch can..."

File: odkx-src/build-scripts.rst [hunk 25]
```diff
@@ -148,14 +148,45 @@ To run the UI tests:
...
+  3. **master**:  **masterBasicRelease** is the build variant that corresponds to a release build of the master branch. **masterBasicDebug** is the build variant that corresponds to a debug build of the master branch.It is the official and stable release of an application, this is the latest release of ODK-X application.
+
+Gradle creates a build variant for every possible combination of the product flavor and build types that you configure. As different code bases is used for each flavor variant.
+
+Steps to change build variants in Android studio:
+
+1. To change the build variant Android Studio uses, select **Build** > **Select Build Variant** in the menu bar.
+2. The Build Variants panel has two columns: **Module** and **Active Build Variant**. The **Active Build Variant** value for the module determines which build variant the IDE deploys to your connected device and is visible in the editor.
```
Comment (elmps2018) [inline odkx-src/build-scripts.rst:178]: What is IDE? might need to define this acronym.

File: odkx-src/build-scripts.rst [hunk 26]
```diff
@@ -148,14 +148,45 @@ To run the UI tests:
...
+    :alt: "Change build variant"
+    :class: device-screen-vertical
 
 .. _build-scripts-internal:
 
 Internal Build Files
 ------------------------
 
-This section covers the files that are stored inside each of the Android projects. These paths follow the same pattern for each Android project, just the project name differs. For clarity, the root level of the project will be referred to as :file:`root` and the app/lib level of the project will be referred to as :file:`app`. So, for example, the path :file:`services/services_app/build.gradle` becomes :file:`project/app/build.gradle`.
+This section covers the files that are stored inside each of the Android projects. These paths follow the same pattern for each Android project, the project name differs. For clarity, the root level of the project will be referred to as :file:`root` and the app/lib level of the project will be referred to as :file:`app`. So, for example, the path :file:`services/services_app/build.gradle` becomes :file:`project/app/build.gradle`.
```
Comment (elmps2018) [inline odkx-src/build-scripts.rst:189]: I think just or something similar should be here, "each Android project, the project name differs" does not scan.

File: odkx-src/build-scripts.rst (hunk 21 above)
Comment (khushishikhu) [reply odkx-src/build-scripts.rst:163]: @elmps2018 Yes it does.

Comment (elmps2018) [reply odkx-src/build-scripts.rst:163]: Please revise to say that instead.

Comment (khushishikhu) [reply odkx-src/build-scripts.rst:163]: @elmps2018 should I drop "a specific build variant" from the sentence?

File: odkx-src/build-scripts.rst [hunk 27]
```diff
@@ -148,7 +148,36 @@ To run the UI tests:
...
+~~~~~~~~~~~~~~~~~~
+
+
+ODK-X has a modular framework design with inter-dependencies between various ODK-X tools. To manage accepting code contributions, bug fixes, and other enhancements that might affect other tools in an unforeseen way ODK-X uses a 3 staged release workflow that involves 3 branches.
+
+- development: the branch where new development, upgrades, and features are contributed and tested
+- demo: This branch is the last stable version of development. This provides a staging area used for testing before moving to an official release. This is where preview releases are staged.
+- master: The current stable release.
+
+Since the branches are at different stages of development the dependency libraries are at different stages of development. To make sure you are building against the correct dependencies you need to either checkout a local copy of the dependencies using the same branch name in the same directory OR you can adjust the build variant to match the dependencies causing Gradle and Android Studio to fetch the correct dependencies when compiling.Each build variant represents a different version of your app that you can build.
```
Comment (linl33) [inline odkx-src/build-scripts.rst:161]: > compiling.Each build variant

Missing space before Each

File: odkx-src/build-scripts.rst [hunk 28]
```diff
@@ -148,7 +148,36 @@ To run the UI tests:
...
+  2. **demo**:  **demoBasicRelease** is the build variant that corresponds to a release build of the demo branch. **demoBasicDebug** is the build variant that corresponds to a debug build of the demo branch. This is the preview release of an application before launching the official release. This branch can be used by project maintainers for testing out the application if it is not creating any errors.
+  3. **master**:  **masterBasicRelease** is the build variant that corresponds to a release build of the master branch. **masterBasicDebug** is the build variant that corresponds to a debug build of the master branch. It is the official and stable release of an application, this is the latest release of ODK-X application.
+
+Gradle creates a build variant for every possible combination of the product flavor and build types that you configure. As different code bases is used for each flavor variant.
+
+Steps to change build variants in Android studio:
```
Comment (linl33) [inline odkx-src/build-scripts.rst:173]: ~~Android studio~~
Android Studio

File: odkx-src/build-scripts.rst [hunk 29]
```diff
@@ -148,7 +148,36 @@ To run the UI tests:
...
+
+1. To change the build variant Android Studio uses, select **Build** > **Select Build Variant** in the menu bar.
+2. The Build Variants panel has two columns: **Module** and **Active Build Variant**. The **Active Build Variant** value for the module determines which build variant the IDE(android studio) deploys to your connected device and is visible in the editor.
```
Comment (linl33) [inline odkx-src/build-scripts.rst:176]: Given that this section only applies to Android Studio, you can replace _**IDE**_ with _**Android Studio**_.

**Review by linl33 — CHANGES_REQUESTED** [review]

**Comment by elmps2018** [comment]
@khushishikhu this looks good now to me! @wbrunette and @linl33 did you have any further comments?

File: odkx-src/spelling_wordlist.txt [hunk 30]
```diff
@@ -91,6 +91,7 @@ esthetics
 ethiopian
 extensibility
 fallback
+flavor
```
Comment (wbrunette) [inline odkx-src/spelling_wordlist.txt:94]: flavor should not need to be added to the list. It was a style check problem disliking the usage of "flavour" in favor of "flavor".

**Review by linl33 — APPROVED** [review]
